src/GraphQLParser/ParserContext.cs

```diff
@@ -37,6 +41,23 @@ public ParserContext(ROM source, ParserOptions options)
             Advance();
         }
 
+        private void IncreaseDepth()
+        {
+            // Encourage compiler inlining of this method by moving exception to a separate method
+            if (_currentDepth++ >= _maxDepth)
+                ThrowMaxDepthException();
+        }
+
+        private void ThrowMaxDepthException()
+        {
+            throw new GraphQLSyntaxErrorException("Maximum depth exceeded.", _source, _currentToken.Start);

```

Comment:
It's not a syntax exception though we _can_ use `_source` and `_currentToken`. `InvalidOperationException` suits better for me.

Reply:
I can create a new exception for it that inherits from `GraphQLSyntaxErrorException`.  Or we can create a new parent exception to cover both child exceptions.  But it should be a parser exception just as exceeding the maximum depth during json parsing throws a `JsonException`.

Also, I would like to maintain binary compatibility within this PR.  Then it can pulled in via nuget by users immediately, fixing the vulnerability within GraphQL.NET.

Reply:
I think we should use `_source` and `_currentToken`.  No reason not to do so.

Reply:
I'm fine with `GraphQLParserException` base class and `GraphQLSyntaxErrorException: GraphQLParserException` + `GraphQLMaxDepthExceededException: GraphQLParserException`.

Reply:
Updated as suggested.

Reply:
Forgot to push?

Reply:
Yup